src/functions/keycloak/update-user-attributes/handler.ts

```diff
@@ -5,76 +5,78 @@ import { CreateSeveraApiService } from "src/services/severa-api-service";
+    const allowedKeys = ["isSeveraOptIn"];
 
-  const allowedKeys = ["isSeveraOptIn"]
+    if (!id || !attributes || typeof attributes !== "object") {
+      return {
+        statusCode: 400,
+        body: JSON.stringify({ message: "Missing or invalid parameters: 'id' or 'attribute'." })
+      };
+    }
 
-  if (!id || !attributes || typeof attributes !== "object") {
-    return {
-      statusCode: 400,
-      body: JSON.stringify({ message: "Missing or invalid parameters: 'id' or 'attribute'." }),
-    };
-  }
+    const keys = Object.keys(attributes);
+    if (!keys.every((key) => allowedKeys.includes(key))) {
+      return {
+        statusCode: 400,
+        body: JSON.stringify({ message: "Attributes contain invalid keys." })
+      };
+    }
 
-  const keys = Object.keys(attributes);
-  if (!keys.every(key => allowedKeys.includes(key))) {
-    return {
-      statusCode: 400,
-      body: JSON.stringify({ message: "Attributes contain invalid keys." }),
-    };
-  }
+    const api = CreateKeycloakApiService();
+    const severaApi = CreateSeveraApiService();
 
-  const api = CreateKeycloakApiService();
-  const severaApi = CreateSeveraApiService()
+    if (!Object.prototype.hasOwnProperty.call(attributes, attributeName)) {
+      return {
+        statusCode: 400,
+        body: JSON.stringify({ message: `Attribute ${attributeName} is missing in request body.` })
+      };
+    }
 
-  let severaUser: {email:string, guid:string } | null = null
+    const severaUser = await severaApi.getUserByEmail(email, attributes);
 
-  if (process.env.NODE_ENV === "development") {
-    severaUser = await severaApi.getTestUser();
-  } else {
-    severaUser = await severaApi.getUserByEmail(email, attributes);
-  }
-  if (!severaUser || !severaUser.email) {
-    return {
-      statusCode: 404,
-      body: JSON.stringify({ message: "Severa user not found." }),
-    };
-  }
+    if (!severaUser?.email || !severaUser?.guid) {
+      return {
+        statusCode: 404,
+        body: JSON.stringify({ message: "Severa user not found." })
+      };
+    }
 
-  if (!attributes.isActive) {
-    attributes.isActive = ["Active"];
-  }
-  attributes["severa-user-id"] = [severaUser.guid];
+    if (!attributes.isActive) {
+      attributes.isActive = ["Active"];
+    }
+    attributes["severa-user-id"] = [severaUser.guid];
 
-  await api.updateUserAttribute(id, attributes);
+    await api.updateUserAttribute(id, attributes);
 
-  await severaApi.getUserByEmail(severaUser.email, attributes);
-  
-  return {
-    statusCode: 200,
-    body: JSON.stringify({ message: "IsSeveraOptin has already been updated in Severa, and severaId will be added to Keycloak." }),
-  };
-} catch (error) {
-  return {
-    statusCode: 500,
-    headers: { "Content-Type": "application/json" },
-    body: JSON.stringify({ message: error.message, stack: error.stack }),
-  };
-}
+    await severaApi.getUserByEmail(severaUser.email, attributes);
```

Comment:
IT's odd you are calling this method again. Did you not break this don't into smaller keycloak methods as I suggested. E.g. one to find a user based on email, one to update the keyword etc. Rather than having them all in the the same method with a somewhat misleading name

Reply:
So now I have gone through the getUserByEmail method and left comments related to that, but can you explain why you are calling this twice in the handler? You should only need to call it once.